### ISSUE TITLE
Long delay when reading Tamil symbol

### ISSUE THREAD

**Issue by feerrenrut** [espeak-ng/espeak-ng#332]
As described in https://github.com/nvaccess/nvda/pull/7751. A (not quite) infinite loop is entered when the ta_emoji file is present and the `ள்` symbol is read. I suspect that this symptom is somewhat build dependent. 

The loop that espeak seems to get stuck in is here: https://github.com/espeak-ng/espeak-ng/blob/master/src/libespeak-ng/dictionary.c#L2589Q

The call stack looks like the following:
```
espeak.dll!LookupDict2(Translator * tr=0x073c9078, const char * word=0x0760c840, const char * word2=0x0760dc46, char * phonetic=0x0760d250, unsigned int * flags=0x0760d34c, int end_flags=2, WORD_TAB * wtab=0x0760df68) Line 2591	C	Symbols loaded.
espeak.dll!LookupDictList(Translator * tr=0x073c9078, char * * wordptr=0x0760d370, char * ph_out=0x0760d250, unsigned int * flags=0x0760d34c, int end_flags=2, WORD_TAB * wtab=0x0760df68) Line 2870	C	Symbols loaded.
espeak.dll!TranslateWord3(Translator * tr=0x073c9078, char * word_start=0x0760dc3f, WORD_TAB * wtab=0x0760df68, char * word_out=0x0760d6e2) Line 611	C	Symbols loaded.
espeak.dll!TranslateWord(Translator * tr=0x073c9078, char * word_start=0x0760dc3f, WORD_TAB * wtab=0x0760df68, char * word_out=0x0760d6e2) Line 1135	C	Symbols loaded.
espeak.dll!TranslateWord2(Translator * tr=0x073c4fa0, char * word=0x0760dc3f, WORD_TAB * wtab=0x0760df68, int pre_pause=0) Line 1459	C	Symbols loaded.
espeak.dll!TranslateClause(Translator * tr=0x073c4fa0, int * tone_out=0x0760f4c8, char * * voice_change=0x0760f4bc) Line 2579	C	Symbols loaded.
espeak.dll!SpeakNextClause(int control=1) Line 1562	C	Symbols loaded.
espeak.dll!Synthesize(unsigned int unique_identifier=0, const void * text=0x0550b5c0, int flags=275) Line 450	C	Symbols loaded.
espeak.dll!sync_espeak_Synth(unsigned int unique_identifier=0, const void * text=0x0550b5c0, unsigned int position=0, espeak_POSITION_TYPE position_type=0, unsigned int end_position=0, unsigned int flags=275, void * user_data=0x00000000) Line 528	C	Symbols loaded.
espeak.dll!espeak_ng_Synthesize(const void * text=0x0550b5c0, unsigned int size=0, unsigned int position=0, espeak_POSITION_TYPE position_type=0, unsigned int end_position=0, unsigned int flags=275, unsigned int * unique_identifier=0x0550b378, void * user_data=0x00000000) Line 636	C	Symbols loaded.
espeak.dll!espeak_Synth(const void * text=0x0550b5c0, unsigned int size=0, unsigned int position=0, espeak_POSITION_TYPE position_type=0, unsigned int end_position=0, unsigned int flags=275, unsigned int * unique_identifier=0x0550b378, void * user_data=0x00000000) Line 91	C	Symbols loaded.
```

The `text` param (in espeak_synth) points at the following unicode string: `<voice xml:lang="en-GB">ள்  </voice>`

Please let me know if there is any more information I can provide to help track down the cause.

**Comment by feerrenrut** [comment]
This is quite possibly related to 'Emoji support produces incomplete or corrupt translations' #308 however, without further investigation I couldn't be certain.

**Comment by rhdunn** [comment]
This has now been fixed in 566e904b332c5816c3b4c02a7cb618c6b217d980. The bug is related to https://github.com/espeak-ng/espeak-ng/issues/287. Specifically, the LookupDict2 code was reading the length of the entry in order to skip it, but was treating that as a signed value, so it was negative in this case which resulted in that loop not terminating.

I have added the character to the tests/translate.test script. This also has tests for https://github.com/nvaccess/nvda/issues/7805, which I am not able to reproduce, so I don't know if it has been fixed or is a result of a build option or compiler that NVDA is using that is different to eSpeak (I have tested on both Linux with gcc and Windows with msvc).
